scripts/gha/test_simulator.py

```diff
@@ -420,13 +418,18 @@ def _get_bundle_id(app_path, config):
       return api["bundle_id"]
 
 
-def _run_apple_gameloop_test(bundle_id, app_path, gameloop_app, device_id):
+def _run_apple_gameloop_test(bundle_id, app_path, gameloop_app, device_id, retry=1):
   """Run gameloop test and collect test result."""
   logging.info("Running apple gameloop test: %s, %s, %s, %s", bundle_id, app_path, gameloop_app, device_id)
   _install_apple_app(app_path, device_id)
   _run_xctest(gameloop_app, device_id)
   logs = _get_apple_test_log(bundle_id, app_path, device_id)
   _uninstall_apple_app(bundle_id, device_id)
+  if retry > 1:
+    results = test_validation.validate_results_cpp(logs)
+    if not results.complete:
+      return _run_apple_gameloop_test(bundle_id, app_path, gameloop_app, device_id, retry=retry-1)
```

Comment:
Is it worth appending something to the logs that notes the number of retries?

Reply:
added